plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -0,0 +1,135 @@
+package splunkmetric
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/influxdata/telegraf"
+)
+
+type serializer struct {
+	TimestampUnits time.Duration
+}
+
+func NewSerializer(timestampUnits time.Duration) (*serializer, error) {
+	s := &serializer{
+		TimestampUnits: truncateDuration(timestampUnits),
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	var serialized string
+
+	m, err := s.createObject(metric)
+	if err == nil {
+		serialized = m + "\n"
+	}
+
+	return []byte(serialized), nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized string
+
+	var objects []string
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err == nil {
```

Comment:
The error should be logged somewhere when not nil. Otherwise metrics are going to be getting dropped and the user won't have any idea why.